engine/common.ftl

```diff
@@ -631,6 +634,74 @@ behaviour.
     [#return {}]
 [/#function]
 
+[#function getProcessorCounts processorProfile multiAz desiredCount="" minCount="" maxCount="" ]
+
+    [#local fixedMaxCount = maxCount?has_content?then(
+                                maxCount,
+                                (proccessorProfile.MaxCount)!""
+                            )]
+    
+    [#local fixedMinCount = minCount?has_content?then(
+                                minCount,
+                                (processorProfile.MinCount)!""
+                            )]
+
+    [#local fixedDesiredCount = desiredCount?has_content?then(
+                                    desiredCount,
+                                    (processorProfile.DesiredCount)!""
+                                )]
+
+    [#if fixedMaxCount?has_content ]
+        [#local maxCount = fixedMaxCount ]
+    [#elseif processorProfile.MaxPerZone?has_content ]
+        [#local maxCount = processorProfile.MaxPerZone ]
+        [#if multiAZ]
+            [#local maxCount = maxCount * zones?size]
+        [/#if]
+    [#else]
+        [@fatal
+            message="Processor profile does not have a MaxCount"
+            context=processorProfile
+        /]
+    [/#if]
+
+    [#if fixedMinCount?has_content ]
+        [#local minCount = fixedMinCount ]
+    [#elseif processorProfile.MinPerZone?has_content ]
+        [#local minCount = (processorProfile.MinPerZone)!1]
+        [#if multiAZ]
+            [#local minCount = minCount * zones?size]
+        [/#if]
+    [#else]
+        [@fatal
+            message="Processor profile does not have a MinCount"
+            context=processorProfile
+        /]
+    [/#if]
+
+    [#if fixedDesiredCount?has_content ]
+        [#local desiredCount = fixedDesirecCount ]
```

Comment:
> fixedDesire**c**Count  :)

Reply:
Fixed, who needs spell checkers when you've got reviewers :)